### PR TITLE
when a field is removed clear its value so rules depending on the rem…

### PR DESCRIPTION
**Pull request by adelhrashed**
when a field is removed clear its value so rules depending on the removed field are run correctly

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -73,3 +73,23 @@ export function listAllFields(rules) {
   });
   return allFields.flat();
 }
+
+export function removeFieldValue(path, formData = {}) {
+  let separator = path.indexOf(".");
+
+  if (separator === -1) {
+    formData[path] = undefined;
+    // set(formData, path, undefined);
```
Comment (Haegin) [inline src/utils.js]: Can this be removed or should it be used?

File: src/utils.js [hunk 2]
```diff
@@ -73,3 +73,23 @@ export function listAllFields(rules) {
   });
   return allFields.flat();
 }
+
+export function removeFieldValue(path, formData = {}) {
+  let separator = path.indexOf(".");
```
Comment (Haegin) [inline src/utils.js:78]: I think rjsf-conditionals uses the selectn library for similar path based lookups - is this path compatible and could we use that here to simplify this code?

Comment (adelhrashed) [reply src/utils.js:78]: selectn library does not automatically look into nested array. For example,
```js
{a:{
 b:{
  c:[{d:2},{d:3}]
  }
}
}
```
The path that will be passed to the function is `a.b.c.d`. However, to actually delete `d`, the should be converted to two paths `a.b.c[0].d` and `a.b.c[1].d`. Those two paths can then be used with set method from lodash. That was first my solution which I found to be more complicated.

File: src/utils.js (hunk 1 above)
Comment (adelhrashed) [reply src/utils.js]: it can be removed

**Review by Haegin — APPROVED** [review]

**Pull request merged**